Ical.Net/Evaluation/RecurrencePatternEvaluator.cs

```diff
@@ -636,15 +638,25 @@ private static IEnumerable<CalDateTime> GetDayVariants(IEnumerable<CalDateTime>
 
     private static IEnumerable<CalDateTime> GetDayVariantsLimited(IEnumerable<CalDateTime> dates, RecurrencePattern pattern)
         =>
-        // If no offset is specified, simply test the day of week!
-        // FIXME: test with offset...
-        dates.Where(date => pattern.ByDay.Any(weekDay => weekDay.DayOfWeek.Equals(date.DayOfWeek)));
-
+            // If no offset is specified, simply test the day of week.
+            // When an offset is present, use GetAbsWeekDays to compute the concrete
+            // weekday dates according to the frequency and check containment.
+            dates.Where(date => pattern.ByDay.Any(weekDay =>
+            {
+                if (weekDay.Offset is null)
+                    return weekDay.DayOfWeek.Equals(date.DayOfWeek);
+
+                // When limiting with an offset (e.g. "22MO" or "1MO"), compute the
+                // absolute dates for that WeekDay in the appropriate scope and
+                // check if the candidate matches one of them.
+                return GetAbsWeekDays(date, weekDay, pattern).Any(d => d.Equals(date));
```

Comment:
We have the same kind of code in the expand variant, both can be pretty expensive. Also the implementations of `GetAbsWeekDays*` could be significantly improved. Candidates for the future. :-)